clients/admin-ui/src/features/dataset/DatasetFieldsTable.tsx

```diff
@@ -1,26 +1,31 @@
 import { Table, Tbody, Td, Th, Thead, Tr } from "@fidesui/react";
 
-import { DatasetField } from "./types";
+import { ColumnMetadata, DatasetField } from "./types";
 
 interface Props {
   fields: DatasetField[];
+  columns: ColumnMetadata[];
 }
 
-const DatasetFieldsTable = ({ fields }: Props) => (
+const DatasetFieldsTable = ({ fields, columns }: Props) => (
   <Table size="sm">
     <Thead>
       <Tr>
-        <Th pl={0}>Field Name</Th>
-        <Th pl={0}>Description</Th>
-        <Th pl={0}>Identifiability</Th>
+        {columns.map((c) => (
+          <Th key={c.name} pl={0}>
+            {c.name}
+          </Th>
+        ))}
       </Tr>
     </Thead>
     <Tbody>
       {fields.map((field) => (
```

Comment:
And fields

Reply:
`fields` is also guaranteed to be an array by typescript. I defined the typescript types based on the backend pydantic types, and so this should be safe